### PR TITLE
add data-astro-rerun to scripts

### PR DESCRIPTION
**Pull request by OliverSpeir**
## Changes

Adds autocomplete for `data-astro-rerun` to scripts

## Testing

YOLO, but I'm more than happy to do some if guidance can be provided

## Related

https://github.com/withastro/docs/pull/6798
https://github.com/withastro/astro/pull/9977

### CODE REVIEW

File: packages/language-server/src/plugins/html-data.ts [hunk 1]
```diff
@@ -91,6 +91,18 @@ export const astroElements = newHTMLDataProvider('astro-elements', {
 						},
 					],
 				},
+				{
+					name: 'data-astro-reload',
```
Comment (martrapp) [inline packages/language-server/src/plugins/html-data.ts:95]:
```suggestion
					name: 'data-astro-rerun',
```

File: .changeset/few-ducks-explain.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@astrojs/language-server': patch
+---
+
+Add auto complete for data-astro-reload on script tags
```
Comment (martrapp) [inline .changeset/few-ducks-explain.md:5]:
```suggestion
Add auto complete for data-astro-rerun on script tags
```

**Comment by martrapp** [comment]
@OliverSpeir You might also want to change the title of the PR 😄

File: .changeset/few-ducks-explain.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@astrojs/language-server': patch
+---
+
+Add auto complete for data-astro-rerun on script tags
```
Comment (VoxelMC) [inline .changeset/few-ducks-explain.md:5]:
```suggestion
Adds auto-complete for `data-astro-rerun` on script tags
```

**Review by VoxelMC — APPROVED** [review]
Thank you, Oliver for adding `data-astro-rerun`. It will solve so many issues that everyone has been having with ViewTransitions! It's definitely going to encourage me to start using ViewTransitions finally.

**Review by martrapp — APPROVED** [review]
LGTM!

**Pull request merged**